numpy/ma/core.py

```diff
@@ -4067,7 +4067,18 @@ def __repr__(self):
             separator=", ",
             prefix=indents['mask'] + 'mask=',
             suffix=',')
-        reprs['fill_value'] = repr(self.fill_value)
+
+        if self._fill_value is None:
+            self.fill_value  # initialize fill_value
+        if self._fill_value.dtype == self.dtype:
```

Comment:
To avoid `np.str_('N/A')` maybe
```
if (self._fill_value.dtype == self.dtype
        or self.dtype.kind in 'US' and self.dtype.kind == self._fill_value.dtype.kind):
```

Reply:
sure why not, done.  Reall, it's nonesense probably that they even _can_ diverge, but since strings are usually longr than the typical `'N/A'` fill value, I don't mind guessing this.